opensearch_py_ml/sentence_transformer_model/sentencetransformermodel.py

```diff
@@ -0,0 +1,677 @@
+        model.save("trained_pytorch_model/")
+        device = "cpu"
+        cpu_model = model.to(device)
+        print(f"Total training time: {time.time() - init_time}")
+
+        for key in out_q.keys():
+            out_q[key] = out_q[key].to(device)
+
+        traced_cpu = torch.jit.trace(
+            cpu_model,
+            (
+                {
+                    "input_ids": out_q["input_ids"],
+                    "attention_mask": out_q["attention_mask"],
+                }
+            ),
+            strict=False,
+        )
+
+        print("Preparing model to save")
+        torch.jit.save(traced_cpu, output_model_path)
+
+        print("Model saved to path: " + output_model_path)
+        return model
+
+    def zip_model(self, model_path: str = None, zip_file_name: str = None) -> None:
+        """
+        Description:
+        zip the model file and its tokenizer.json file to prepare to upload to the Open Search cluster
+
+        Parameters:
+        model_path: str
+            Optional, path to find the model file, if None, default as trained_model.pt file in current path
+        zip_file_name: str =None
+            Optional, file name for zip file. if None, default as zip_model.zip
+
+        Return:
+            None
+        """
+
+        if model_path is None:
+            model_path = os.path.join(os.getcwd(), "trained_model.pt")
+
+        if zip_file_name is None:
+            zip_file_name = "zip_model.zip"
+
+        if not os.path.exists("trained_pytorch_model/tokenizer.json"):
+            raise Exception(
+                "Cannot find tokenizer.json in trained_pytorch_model/tokenizer.json"
+            )
+        if not os.path.exists(model_path):
+            raise Exception("Cannot find model in the model path")
+
+        # Create a ZipFile Object
+        with ZipFile("zip_model.zip", "w") as zipObj:
+            zipObj.write(model_path)
+            zipObj.write("trained_pytorch_model/tokenizer.json")
+        print("zip file is saved to" + os.getcwd() + "/" + zip_file_name)
+
+    def set_up_accelerate_config(
+        self,
+        compute_environment: str = None,
+        num_machines: int = None,
+        num_processes: int = None,
+    ) -> None:
+        """
+        Description:
+        get default config setting based on the number of GPU on the machine
+        if users require other configs, users can run !acclerate config for more options.
+
+        Parameters:
+        compute_environment: str
+            optional, compute environment type to run model, if None, default using 'LOCAL_MACHINE'
+        num_machines: int
+            optional, number of machine to run model , if None, default using 1
+
+        Return:
+            None
+        """
+
+        if compute_environment is None or compute_environment == 0:
+            compute_environment = "LOCAL_MACHINE"
+        else:
+            subprocess.run("accelerate config")
+            return
+
+        if num_machines is None:
```

Comment:
put this as default variable: `num_machines: int = 1,`

Reply:
assigned in the default value